ktor-network/js/src/io/ktor/network/sockets/nodejs/node.net.js.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.sockets.nodejs
+
+import io.ktor.network.sockets.*
+import org.khronos.webgl.*
+
+internal actual fun nodeNet(): NodeNet? = js("eval('require')('node:net')").unsafeCast<NodeNet?>()
```

Comment:
Won't it lead to [KTOR-6158](https://youtrack.jetbrains.com/issue/KTOR-6158)?

Reply:
I'm okay with the first option if we create an issue to track compatibility of network module with ES and add this context there. @e5l what do you think?

Reply:
yep, the first option looks good

Reply:
@whyoleg, could you [create an issue](https://youtrack.jetbrains.com/newIssue?project=KTOR) and rebase this branch one more time? :)

Reply:
created https://youtrack.jetbrains.com/issue/KTOR-7659/Make-ktor-network-compatible-with-ES-modules-for-nodejs
and rebased